### PR TITLE
#445 Updating an inbound

### LINKED ISSUE #445: Add delete line functionality to inbound

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #445 and #435

- Adds updating of an invoice features
- Few off roads that i'll point out!

### CODE REVIEW

File: packages/common/src/intl/IntlProvider.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ export const IntlProvider: React.FC = ({ children }) => {
     const minuteInMilliseconds = 60 * 1000;
     const expirationTime =
       process.env['NODE_ENV'] === 'development'
-        ? 5 * minuteInMilliseconds
+        ? 0
```
Comment (joshxg) [inline packages/common/src/intl/IntlProvider.tsx:15]: I got fooled a couple times already by localisations not updating 😁  I think this works when you refresh without having to clear local storage

File: packages/common/src/intl/locales/en/common.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
   "app.sync": "Sync",
   "app.tools": "Tools",
   "app.welcome": "Hello, {name}!",
-  "breadcrumb.item": "#{id}",
+  "breadcrumb.item": "#{{id}}",
```
Comment (joshxg) [inline packages/common/src/intl/locales/en/common.json:22]: just offroad translation fix

File: packages/host/src/bootstrap.tsx [hunk 3]
```diff
@@ -9,7 +9,7 @@ if (process.env.NODE_ENV === 'development') {
   } = require('@openmsupply-client/mock-server/src/worker/client');
   const worker = setupMockWorker();
 
-  // worker.start();
+  worker.start();
```
Comment (joshxg) [inline packages/host/src/bootstrap.tsx:12]: I thought I did this already 🤔

File: packages/invoices/src/InboundShipment/DetailView/Footer.tsx [hunk 4]
```diff
@@ -26,7 +26,7 @@ interface InboundDetailFooterProps {
   save: () => Promise<void>;
 }
 
-const createStatusLog = () => {
+const createStatusLog = (status: 'DRAFT' | 'CONFIRMED' | 'FINALISED') => {
```
Comment (joshxg) [inline packages/invoices/src/InboundShipment/DetailView/Footer.tsx:29]: just an offroad fix for statuses being funny

File: packages/invoices/src/InboundShipment/DetailView/Footer.tsx [hunk 5]
```diff
@@ -58,78 +73,80 @@ export const Footer: FC<InboundDetailFooterProps> = ({ draft, save }) => {
...
+            height={64}
+          >
+            <ToggleButton
+              disabled={!isInboundEditable(draft)}
+              value={!!draft.onHold}
+              selected={!!draft.onHold}
+              onClick={(_, value) => {
+                draft.update && draft.update('q', !value);
+              }}
+              label={t('label.hold')}
```
Comment (joshxg) [inline packages/invoices/src/InboundShipment/DetailView/Footer.tsx:91]: this is the main change - from `hold` to `onHold`- if this was typed better 😢

File: packages/invoices/src/InboundShipment/DetailView/reducer/reducer.ts [hunk 6]
```diff
@@ -197,9 +203,18 @@ export const reducer = (
...
           }, draft.items);
 
+          break;
+        }
+        case ActionType.UpdateInvoice: {
+          const { payload } = action;
+          const { key, value } = payload;
+
+          state.draft[key] = value;
+
```
Comment (joshxg) [inline packages/invoices/src/InboundShipment/DetailView/reducer/reducer.ts:213]: just needed to add this to the draft. Wish there was a way to ensure the fn was non-optional

File: packages/invoices/src/InboundShipment/ListView/Toolbar.tsx [hunk 7]
```diff
@@ -57,7 +57,7 @@ export const Toolbar: FC<{
       }}
     >
       <InputWithLabelRow
-        label={'label.search'}
+        label={t('label.search')}
```
Comment (joshxg) [inline packages/invoices/src/InboundShipment/ListView/Toolbar.tsx:60]: just offroad translation fix

File: packages/invoices/src/OutboundShipment/DetailView/Footer.tsx [hunk 8]
```diff
@@ -86,7 +86,7 @@ export const Footer: FC<OutboundDetailFooterProps> = ({ draft, save }) => {
               value={!!draft.onHold}
               selected={!!draft.onHold}
               onClick={(_, value) => {
-                draft.update?.('hold', !value);
+                draft.update?.('onHold', !value);
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/DetailView/Footer.tsx:89]: the fix for outbound also

File: packages/common/src/intl/IntlProvider.tsx (hunk 1 above)
Comment (mark-prins) [reply packages/common/src/intl/IntlProvider.tsx:15]: yah, caching 🙄

**Review by mark-prins — APPROVED** [review]
👍  nice!

**Pull request merged**